### PR TITLE
Update Xcodeproj

### PR DESCRIPTION
**Pull request by adellibovi**
Resolves https://github.com/tuist/tuist/issues/2650

### Short description 📝

Hello,
we are updating Xcodeproj, the version 8.1.0 includes some performances optimisation when generating the projects. It is especially evident on machine with few cores.

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase.
- [x] The changes have been tested following the [documented guidelines](https://docs.tuist.io/contributors/testing-strategy/).
- [x] The `CHANGELOG.md` has been updated to reflect the changes. In case of a breaking change, it's been flagged as such.
- [x] In case the PR introduces changes that affect users, the documentation has been updated.

### CODE REVIEW

**Review by danyf90 — APPROVED** [review]

File: Package.swift [hunk 1]
```diff
@@ -51,7 +51,7 @@ let package = Package(
         ),
     ],
     dependencies: [
-        .package(url: "https://github.com/tuist/XcodeProj.git", .upToNextMajor(from: "8.0.0")),
+        .package(url: "https://github.com/tuist/XcodeProj.git", .upToNextMajor(from: "8.2.0")),
```
Comment (fortmarek) [inline Package.swift:54]: There's also a `Project.swift` that you need to update as well. We'd have liked to remove `Package.swift` completely in the future but for now we have to do so manually.

Comment (adellibovi) [reply Package.swift:54]: Done @fortmarek

Comment (fortmarek) [reply Package.swift:54]: Sorry, but you'll need to also regenerate the project, so `.package.resolved` gets updated 🙈

Comment (adellibovi) [reply Package.swift:54]: Done 😛 @fortmarek

Comment (fortmarek) [reply Package.swift:54]: Thanks!!

**Review by fortmarek — APPROVED** [review]

**Review by laxmorek — APPROVED** [review]

**Pull request merged**